File: signature/openssl.go
package signature

var OpenSSLSignature = Signature{
	Name:        "OpenSSL",
	Description: "OpenSSL is a software library for applications that provide secure communications over computer networks.",
	Cpe:         "cpe:/a:openssl:openssl",

	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "OpenSSL",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `OpenSSL/(\d+\.\d+\.\d+[a-zA-Z]?)`,
		}},
	},

	OnlySameHost: true,
}

File: src/signatures/technologies/openssl.ts
import type { Signature } from "../_types.js";

export const openSslSignature: Signature = {
  name: "OpenSSL",
  description:
    "OpenSSL is a software library for applications that secure communications over computer networks against eavesdropping or need to identify the party at the other end.",
  cpe: "cpe:/a:openssl:openssl",
  rule: {
    confidence: "high",
    headers: {
      Server: "OpenSSL(?:/([\\d.]+[a-z]?))?",
    },
  },
};
